conans/client/loader.py

```diff
@@ -373,6 +373,8 @@ class defining the Recipe, but also process possible existing generators
                 raise ConanException("More than 1 conanfile in the file")
         elif issubclass(attr, Generator) and attr != Generator:
             generator_manager.add(attr.__name__, attr, custom=True)
+        elif attr.__name__ == "Recipe":
+            result = type('Recipe', (attr, ConanFile, object), {})
```

Comment:
This seems to work, I have run some tests forcing ``GenConanfile`` to always use this way, and the only tests the break seems the ones that change the recipe revision (obviously, conanfile.py is different).

However, this feels a bit like too magic. I am creating a completely new type, ``Recipe`` that derives from the user ``Recipe`` class written in the ``conanfile.py`` (but still a different entity, I am not just injecting the ``ConanFile`` base class).

So not sure what would be the best approach:
- To use explicit ``from conan import ConanFile  & class Pkg(ConanFile)``
- To use magical ``class Recipe:``